src/graphics/styles/codac2_Color.cpp

```diff
@@ -12,46 +12,162 @@
+  assert(hex_str.size() == 7 || hex_str.size() == 9);
+  assert(hex_str[0] == '#');
+
+  m = Model::RGB;
 
   int red,green,blue,a;
-  std::istringstream(hex_str_.substr(1,2)) >> std::hex >> red;
-  std::istringstream(hex_str_.substr(3,2)) >> std::hex >> green;
-  std::istringstream(hex_str_.substr(5,2)) >> std::hex >> blue;
-  r = (float)red/255.;
-  g = (float)green/255.;
-  b = (float)blue/255.;
+  std::istringstream(hex_str.substr(1,2)) >> std::hex >> red;
+  std::istringstream(hex_str.substr(3,2)) >> std::hex >> green;
+  std::istringstream(hex_str.substr(5,2)) >> std::hex >> blue;
+  (*this)[0] = (float) (red);
+  (*this)[1] = (float) (green);
+  (*this)[2] = (float) (blue);
 
   // Alpha (transparency) component may be appended to the #hexa notation.
-  // Value is '1' (max opacity) by default.
-  if(hex_str_.size() == 9)
+  // Value is '255.' (max opacity) by default.
+  if(hex_str.size() == 9)
   {
-    std::istringstream(hex_str_.substr(7,2)) >> std::hex >> a;
-    alpha = (float)a/255.;
+    std::istringstream(hex_str.substr(7,2)) >> std::hex >> a;
+    (*this)[3] = (float) (a);
+  }
+  else
+    (*this)[3] = 255.;
+}
+
+Color Color::rgb() const
+{
+  if (m==Model::RGB)
+    return *this;
+  else
+  {
+    float r = 0., g = 0., b = 0.;
+
+    // Normalisation des valeurs
+    float h = (*this)[0] / 360.; // Hue normalisée (0 à 1)
+    float s = (*this)[1] / 100.; // Saturation normalisée (0 à 1)
+    float v = (*this)[2] / 100.; // Value normalisée (0 à 1)
+
+    int i = static_cast<int>(h * 6);
+    float f = (h * 6) - i;
+    i = i % 6;
+
+    float p = v * (1 - s);
+    float q = v * (1 - f * s);
+    float t = v * (1 - (1 - f) * s);
+
+    switch (i) {
+        case 0: r = v; g = t; b = p; break;
+        case 1: r = q; g = v; b = p; break;
+        case 2: r = p; g = v; b = t; break;
+        case 3: r = p; g = q; b = v; break;
+        case 4: r = t; g = p; b = v; break;
+        case 5: r = v; g = p; b = q; break;
+    }
+
+    // Conversion vers l'échelle [0, 255]
+    r *= 255.;
+    g *= 255.;
+    b *= 255.;
+
+    return Color({r, g, b,(float) ((*this)[3]*2.55)},Model::RGB);
   }
+}
+
+Color Color::hsv() const
+{
+  if (m==Model::HSV)
+    return *this;
+  else
+  {
+    float r = (*this)[0]/255.;
+    float g = (*this)[1]/255.;
+    float b = (*this)[2]/255.;
+    float c_max = std::max({r, g, b});
+    float c_min = std::min({r, g, b});
+    float delta = c_max - c_min;
+
+    float h = 0.0;
+    if (delta != 0) {
+        if (c_max == r) {
+            h = fmod((g - b) / delta, 6.0);
+        } else if (c_max == g) {
+            h = (b - r) / delta + 2.0;
+        } else if (c_max == b) {
+            h = (r - g) / delta + 4.0;
+        }
+        h /= 6.0;
+        if (h < 0) {
+            h += 1.0;
+        }
+    }
+
+    float s = (c_max == 0) ? 0 : (delta / c_max);
+
+    float v = c_max;
+
+    h*=360.;
+    s*=100.;
+    v*=100.;
+
+    return Color({h, s, v,(float) ((*this)[3]/2.55)},Model::HSV);
```

Comment:
Certains tests semblent ne pas passer sur Mac ou Windows, je pense que c'est lié à cette ligne.
Tu peux essayer avec : 
`return Color({h, s, v,std::min<float>(100.,((*this)[3]/2.55))},Model::HSV);`